src/app/charts/line-graph.component.ts

```diff
@@ -42,121 +42,117 @@ export class LineGraphComponent {
   }
 
   /* Will Update on every @Input change */
   ngOnChanges(): void {
-    if (!this.data || this.data.length === 0) return;
-    this.filterData();
-    this.setup();
-    this.buildSVG();
-    this.setAxisScales();
-    this.drawXAxis();
-    this.drawYAxis();
-    this.populate();
-    this.drawTrendLine();
-    // Temporarily here to show chart control statuses
-    console.log("Min: " + this.min, this.minVal + "; Max: " + this.max, this.maxVal);
+    if (!self.data || self.data.length === 0) return;
+    self.filterData();
+    self.setup();
+    self.buildSVG();
+    self.setAxisScales();
+    self.drawXAxis();
+    self.drawYAxis();
+    self.populate();
+    self.drawTrendLine();
+    self.drawMinMax();
   }
 
   private filterData(): void {
-    var indicator = this.indicator;
+    var indicator = self.indicator;
     // Preserves parent data by fresh copying indicator data that will undergo processing
-    this.extractedData = _.cloneDeep(_.find(this.data, function(obj) {
+    self.extractedData = _.cloneDeep(_.find(self.data, function(obj) {
       return obj["indicator"] == indicator;
     }));
-    _.has(this.extractedData, "data")? this.extractedData=this.extractedData["data"] : this.extractedData=[];
+    _.has(self.extractedData, "data")? self.extractedData=self.extractedData["data"] : self.extractedData=[];
+    // Remove empty day in non-leap years
+    if (self.extractedData[365] && self.extractedData[365]['date'] == null) {
+        self.extractedData.pop(365);
+    }
     // Parse out data by axis for ease of use later
-    this.yData = _.map(this.extractedData, function(d) {
+    self.yData = _.map(self.extractedData, function(d) {
       return d.value;
     });
   }
 
   /* Will setup the chart basics */
   private setup(): void {
-    this.margin = { top: 20, right: 20, bottom: 40, left: 40 };
-    this.width = $('.chart').width() - this.margin.left - this.margin.right;
-    this.height = 200 - this.margin.top - this.margin.bottom;
-    this.xScale = D3.scaleTime().range([0, this.width]);
-    this.yScale = D3.scaleLinear().range([this.height, 0]);
+    self.margin = { top: 20, right: 20, bottom: 40, left: 40 };
+    self.width = $('.chart').width() - self.margin.left - self.margin.right;
+    self.height = 200 - self.margin.top - self.margin.bottom;
+    self.xScale = D3.scaleTime().range([0, self.width]);
+    self.yScale = D3.scaleLinear().range([self.height, 0]);
   }
 
   /* Will build the SVG Element */
   private buildSVG(): void {
-    this.host.html('');
-    this.svg = this.host.append('svg')
-      .attr('width', this.width + this.margin.left + this.margin.right)
-      .attr('height', this.height + this.margin.top + this.margin.bottom)
-      .append('g')
-      .attr('transform', 'translate(' + this.margin.left + ',' + this.margin.top + ')');
+    self.host.html('');
+    self.svg = self.host.append('svg')
+                .attr('width', self.width + self.margin.left + self.margin.right)
```

Comment:
Inconsistent style here. These continuing lines should only be indented by two spaces